public/app/plugins/panel/stat/common.ts

```diff
@@ -107,3 +111,55 @@ export function addOrientationOption<T extends SingleStatBaseOptions>(
     defaultValue: VizOrientation.Auto,
   });
 }
+
+export function formatDisplayValuesWithCustomUnits(
+  fieldValues: FieldDisplay[],
+  customUnits: { customPrefix: string; customSuffix: string }
+): FieldDisplay[] {
+  const { customPrefix, customSuffix } = customUnits;
+
+  // If there are no custom units, do nothing and return the `fieldValues`
+  if (!customPrefix && !customSuffix) {
+    return fieldValues;
+  }
+
+  return fieldValues.map((fieldValue) => {
+    const { fieldType, display } = fieldValue;
+    // `FieldType.number` is the only type on which unit formatting is enforced
+    if (fieldType === FieldType.number) {
+      // Update prefix; prepend custom prefix
+      const previousPrefix = display.prefix ?? '';
+      const updatedPrefix = customPrefix + previousPrefix;
+      // Update suffix; append custom suffix
+      const previousSuffix = display.suffix ?? '';
+      const updatedSuffix = previousSuffix + customSuffix;
+      // Put everything back together
+      const updatedDisplay = { ...display, prefix: updatedPrefix, suffix: updatedSuffix };
+      return { ...fieldValue, display: updatedDisplay };
+    }
+    return fieldValue;
+  });
+}
+
+export function addCustomUnitTextInputs(builder: FieldConfigEditorBuilder<FieldConfig>) {
+  const category = ['Custom units'];
+  return builder
+    .addTextInput({
+      path: 'customPrefix',
```

Comment:
We should add the new option to something like:
https://github.com/grafana/grafana/blob/main/public/app/plugins/panel/candlestick/panelcfg.cue#L27

something like:
```
PanelFieldConfig: {
  prefix?: string		
  suffix?: string					
} @cuetsy(kind="interface")
```

Given that it is already in "custom" i don't think we need/want it to be `custom.customPrefix`

Reply:
Yep, good catch. On it.